pkg/network/server.go

```diff
@@ -889,7 +889,7 @@ func (s *Server) handleVersionCmd(p Peer, version *payload.Version) error {
 
 // handleBlockCmd processes the block received from its peer.
 func (s *Server) handleBlockCmd(p Peer, block *block.Block) error {
-	if s.syncHeaderFetcher.IsActive() || s.syncBlockFetcher.IsActive() || s.blockFetcher.IsActive() || s.syncStateFetcher.IsActive() {
+	if s.config.NeoFSStateSyncExtensions && s.stateSync.IsActive() {
```

Comment:
https://github.com/nspcc-dev/neo-go/pull/3917#discussion_r2113785089 was only about statesync-related fetchers, there's nothing about `s.blockFetcher.IsActive()`. So `s.blockFetcher.IsActive()` is still required here.